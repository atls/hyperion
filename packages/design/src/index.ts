export * from './Sample'
export * from './Layout'
export * from './Container'
export * from './List'
export * from './Text'
export * from './Space'
