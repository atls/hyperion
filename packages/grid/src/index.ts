export * from './Column'
export * from './Grid'
