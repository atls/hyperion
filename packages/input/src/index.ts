export * from './Textarea'
export * from './Input'
export * from './Checkbox'
