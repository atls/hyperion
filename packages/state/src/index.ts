export * from './withState'
export * from './useActive'
export * from './useHover'
export * from './useFocus'
