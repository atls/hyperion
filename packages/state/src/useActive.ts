import { useState } from 'react'

export const useActive = () => {
  const [active, setActive] = useState(false)

  return {
    active,
    onMouseDown: () => setActive(true),
    onMouseUp: () => setActive(false),
  }
}
