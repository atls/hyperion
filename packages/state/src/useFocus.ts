import { useState } from 'react'

export const useFocus = () => {
  const [focus, setFocus] = useState(false)

  return {
    focus,
    onFocus: () => setFocus(true),
    onBlur: () => setFocus(false),
  }
}
