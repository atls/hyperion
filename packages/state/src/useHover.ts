import { useState } from 'react'

export const useHover = () => {
  const [hover, setHover] = useState(false)

  return {
    hover,
    onMouseEnter: () => setHover(true),
    onMouseLeave: () => setHover(false),
  }
}
