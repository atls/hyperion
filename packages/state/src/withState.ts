import { createElement, forwardRef } from 'react'

export const withState = (stateHooks = []) => Target =>
  forwardRef((props, ref) => {
    const state = stateHooks.reduce((result, hook) => ({ ...result, ...hook() }), {})

    return createElement(Target, {
      ...state,
      ...props,
      ref,
    })
  })
