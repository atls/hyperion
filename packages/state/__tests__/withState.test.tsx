import renderer                                     from 'react-test-renderer'
import React, { useEffect }                         from 'react'

import { useActive, useFocus, useHover, withState } from '../src/index'

it('withState should provide all hocs to the component', () => {
  const TargetButton = ({ active, focus, hover, onMouseEnter }) => {
    useEffect(() => {
      onMouseEnter()
    })

    return <button type='submit'>{`${active}, ${focus} and ${hover}`}</button>
  }

  const EnhancedButton = withState([useActive, useFocus, useHover])(TargetButton)

  const tree = renderer.create(<EnhancedButton />)

  expect(tree.toJSON()).toMatchSnapshot()
})
