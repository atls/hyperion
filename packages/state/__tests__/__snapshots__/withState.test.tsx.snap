// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withState should provide all hocs to the component 1`] = `
<button
  type="submit"
>
  false, false and false
</button>
`;
